### ISSUE TITLE
How to force using predefined classes?

### ISSUE THREAD

**Issue by off99555** [tzutalin/labelImg#239]
For example, if I want to have only class `car` and `license plate` and no misspellings are allowed like `ccar`.
I feel like this should be normally forced because it is so sensitive but why it's so easy to put incorrect name when annotating?

**Comment by off99555** [comment]
I want to use this software with `darkflow`.

**Comment by awilliamson** [comment]
`https://github.com/tzutalin/labelImg/blob/master/data/predefined_classes.txt` is what you're looking for? When labelling it should ask you to tick one of the predefined classes.

**Comment by off99555** [comment]
Yes, it asked me but it also provides the option to type it out. I want to give this to labors to label and we can't guarantee that they will click on the predefined class.

**Comment by awilliamson** [comment]
I believe users are more likely to click on the list which pops up, than to type in themselves. There will always be a degree of mislabelling. Providing adequate instruction beforehand should be key.

**Comment by jeffggardner** [comment]
I submitted a PR not sure if it will be accepted
https://github.com/tzutalin/labelImg/pull/263

**Comment by vdalv** [comment]
I agree w/ awilliamson. If these labels are really mission critical, you can write a simple python script that reads through the xml files and detects any label names that aren't "car" or "license plate", then you can go in and fix those.

**Comment by off99555** [comment]
It doesn't fix the core issue. It's like you are coping with it instead of providing an option to never make mistake in the first place. 
There should be an option to force user to not click ok and that should prevent things from breaking better.

**Comment by vdalv** [comment]
Yeah, that's a fair statement.

The UX of this program is nowhere near perfect, and you're more than welcome to fork the repo and improve it as you see fit. FWIW, your current solution sounds pretty reasonable.
